src/storage/v2/edge_accessor.cpp

```diff
@@ -129,10 +129,12 @@ Result<storage::PropertyValue> EdgeAccessor::SetProperty(PropertyId property, co
 
   if (edge_.ptr->deleted) return Error::DELETED_OBJECT;
   using ReturnType = decltype(edge_.ptr->properties.GetProperty(property));
-  std::optional<ReturnType> current_value;
+  std::optional<ReturnType> current_value = edge_.ptr->properties.GetProperty(property);
+  if (current_value == value && !storage_->config_.salient.items.delta_on_identical_property_update) {
```

Comment:
Reverse the order of the `&&` to do the cheaper bool check first.

Reply:
FYI still needs to do `GetProperty` inside the `utils::AtomicMemoryBlock` as it can cause allocation